main_test.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/ONSdigital/dp-permissions-api/features/steps"
+	"os"
+	"testing"
+
+	componenttest "github.com/ONSdigital/dp-component-test"
+	"github.com/cucumber/godog"
+	"github.com/cucumber/godog/colors"
+)
+
+// Mongo version here is overridden in the pipeline by the URL provided in the component.sh
+const MongoVersion = "4.0.23"
+const DatabaseName = "testing"
+
+var componentFlag = flag.Bool("component", false, "perform component tests")
+
+type ComponentTest struct {
+	MongoFeature *componenttest.MongoFeature
+}
+
+func (f *ComponentTest) InitializeScenario(ctx *godog.ScenarioContext) {
+	authorizationFeature := componenttest.NewAuthorizationFeature()
+	topicComponent, err := steps.NewPermissionsComponent(f.MongoFeature)
+	if err != nil {
+		panic(err)
```

Comment:
I think you should return this to Test_main and exit there - its cleaner to have a single exit point in the code IMO.

Reply:
we cannot return error in `InitializeScenario` as this function is passed to godoc.Testsuite to initializeScenarios, we need to follow the same method signature as expected godoc.Testsuite.
Usage of this method :
```
status = godog.TestSuite{
			Name:                 "feature_tests",
			ScenarioInitializer:  f.InitializeScenario,
			TestSuiteInitializer: f.InitializeTestSuite,
			Options:              &opts,
		}.Run()
```